utils/utils.c

```diff
@@ -280,12 +281,87 @@ void fetchGlfsVolServerFromEnv()
   LOG("mgmt", GB_LOG_INFO, "Block Hosting Volfile Server Set to: %s", gbConf.volServer);
 }
 
+
 static int
-initLogDirAndFiles(char *newLogDir)
+glusterLogrotateConfigSet(const char *dom, char *logDir)
+{
+  char *buf = NULL, *line = NULL, *p;
+  int ret, m, len;
+  size_t n;
+  FILE *fp;
+
+  fp = fopen(GB_LOGROTATE_PATH, "r+");
+  if (fp == NULL) {
+    ret = -errno;
+    LOG(dom, GB_LOG_ERROR, "Failed to open file '%s', %m\n", GB_LOGROTATE_PATH);
+    return ret;
+  }
+
+  ret = fseek(fp, 0L, SEEK_END);
+  if (ret == -1) {
+    ret = -errno;
+    LOG(dom, GB_LOG_ERROR, "Failed to seek file '%s', %m\n", GB_LOGROTATE_PATH);
+    goto error;
+  }
+
+  len = ftell(fp);
+  if (len == -1) {
+    ret = -errno;
+    LOG(dom, GB_LOG_ERROR, "Failed to get the length of file '%s', %m\n", GB_LOGROTATE_PATH);
+    goto error;
+  }
+
+  /* to make sure we have enough size */
+  len += strlen(logDir) + 1;
+  if (GB_ALLOC_N(buf, len) < 0) {
+    ret = -ENOMEM;
+    goto error;
+  }
+
+  p = buf;
+  fseek(fp, 0L, SEEK_SET);
+  while ((m = getline(&line, &n, fp)) != -1) {
+    if (strstr(line, "*.log") && strchr(line, '{')) {
+      m = sprintf(p, "%s/*.log {\n", logDir);
+    } else {
+      m = sprintf(p, "%s", line);
+    }
+    if (m < 0) {
+      ret = m;
+      goto error;
+    }
+
+    p += m;
+  }
+  *p = '\0';
+  len = p - buf;
+
+  fseek(fp, 0L, SEEK_SET);
+  truncate(GB_LOGROTATE_PATH, 0L);
+  ret = fwrite(buf, 1, len, fp);
+  if (ret != len) {
+    LOG(dom, GB_LOG_ERROR, "Failed to update '%s', %m\n", GB_LOGROTATE_PATH);
+    goto error;
+  }
```

Comment:
I have not yet tested, but I assume this will not override the first line:
```
$ cat /etc/logrotate.d/gluster-block
# Rotate all the gluster-block logs                                              
/var/log/gluster-block/*.log {         
```

Reply:
Yeah, it won't. I will only replace the second line above.
Thanks.